### PR TITLE
azure: allow anonymous reads when account key is omitted

### PR DESCRIPTION
**Pull request by ry**
cc @benjamin-bader

### CODE REVIEW

File: src/compiler/clang.rs [hunk 1]
```diff
@@ -272,7 +272,14 @@ mod test {
             "-debug-info-kind=constructor"
         );
         assert_eq!(
-            ovec!["-Xclang", "-mllvm", "-Xclang", "-instcombine-lower-dbg-declare=0", "-Xclang", "-debug-info-kind=constructor"],
```
Comment (benjamin-bader) [inline src/compiler/clang.rs]: Did you intend to include this change?  It looks fine but seems unrelated to the rest of the PR.

**Comment by benjamin-bader** [comment]
This looks good to me!

File: src/compiler/clang.rs (hunk 1 above)
Comment (ry) [reply src/compiler/clang.rs]: It's caused by "cargo fmt" - I will revert it

**Comment by froydnj** [comment]
Thanks!
